Automattic/jetpack
# Revert "Frontend email nag for custom domain "

Reverts Automattic/jetpack#33390

As per the Nov 30 Walkthrough discussion, let's remove this as it feels a bit too much.


#### Testing instructions

1. Run `bin/jetpack-downloader test jetpack-mu-wpcom-plugin revert-33390-add/domains-email-nag` on your sandbox
2. Buy a domain with an account that has an unverified email address and map the domain to a site
3. Go to the site's frontend while logged in and confirm you no loner see a nag.
